ic-agent/src/agent/http_transport.rs

```diff
@@ -43,7 +45,15 @@ impl ReqwestHttpReplicaV2Transport {
 
         Ok(Self {
             url: reqwest::Url::parse(&url)
-                .and_then(|url| url.join("api/v2/"))
+                .and_then(|mut url| {
+                    // rewrite *.ic0.app to ic0.app
+                    if let Some(domain) = url.domain() {
+                        if domain.ends_with(IC0_DOMAIN) && domain != IC0_DOMAIN {
```

Comment:
any concerns about case sensitivity?

Reply:
Thankfully `domain`s are all encoded as lowercase as per IDNA spec, so case sensitivity is not a problem. I've added some tests to cover this just to make doubly sure.